Development environment/Docker.md

```diff
@@ -1,56 +1,155 @@
-      MYSQL_ROOT_PASSWORD: root
-      MYSQL_DATABASE: db_name
-      MYSQL_USER: root
-      MYSQL_PASSWORD: root
+      MYSQL_RANDOM_ROOT_PASSWORD: 1
+      MYSQL_DATABASE: wordpress-handbook-app
+      MYSQL_USER: wp
+      MYSQL_PASSWORD: wp
+    ports:
+      - "33066:3306"
+
   phpmyadmin:
     image: phpmyadmin/phpmyadmin
+    container_name: wp-docker-phpmyadmin
+    external_links:
+      - db
+    ports:
+      - 8181:80
     environment:
-      MYSQL_ROOT_PASSWORD: root
+      MYSQL_RANDOM_ROOT_PASSWORD: 1
+      MYSQL_USERNAME: root
+
+  nginx:
+    image: nginx
+    container_name: wp-docker-nginx
+    restart: always
+    volumes:
+      - ./:/var/www/html
+    depends_on:
+      - app
+    volumes:
+      - ./bin/nginx.conf:/etc/nginx/conf.d/default.conf
+      - .:/var/www/html
+      - ./logs:/var/log/nginx
     ports:
-      - 3306:80
+      - 8010:80
+
 volumes:
-  data: {}
+  db_data:
+
 ```
 
-After adding that file and changing the database name, run
+Besides adding `docker-compose.yml` file, create `bin` folder with two files in it: `nginx.conf` and `php.ini` for your nginx and php settings. The `nginx.conf` looks like this
 
-`docker-compose up -d`
+```bash
+## secure headers
+# https://www.owasp.org/index.php/OWASP_Secure_Headers_Project#xxxsp
+add_header X-Xss-Protection "1; mode=block" always;
+# https://www.owasp.org/index.php/OWASP_Secure_Headers_Project#xfo
+add_header X-Frame-Options "SAMEORIGIN" always;
+# https://www.owasp.org/index.php/OWASP_Secure_Headers_Project#xcto
+add_header X-Content-Type-Options "nosniff" always;
+
+## global fastcgi config
+fastcgi_hide_header X-Powered-By;
+fastcgi_pass_header Authorization;
+
+server {
+    listen 80;
+
+    add_header Strict-Transport-Security "max-age=31536000" always;
+
+    root /var/www/html;
+    index index.php;
+
+    access_log /var/log/nginx/access.log;
+    error_log /var/log/nginx/error.log;
+
+    client_max_body_size 64M;
+
+    location / {
+        try_files $uri $uri/ /index.php?$args;
+    }
+
+    location ~ \.php$ {
+        try_files $uri =404;
+        fastcgi_split_path_info ^(.+\.php)(/.+)$;
+        fastcgi_pass app:9000;
+        fastcgi_index index.php;
+        include fastcgi_params;
+        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
+        fastcgi_param PATH_INFO $fastcgi_path_info;
+    }
+}
+```
+
+And `php.ini` looks like this
+
+```bash
+file_uploads = On
+memory_limit = 512M
+upload_max_filesize = 64M
+post_max_size = 64M
+max_execution_time = 600
+```
 
-This will build a Docker container and image with the specified settings from the `.yml` file. Alternatively, you can run
+After adding those files, from your terminal, run the following command
 
-`docker-compose up -d && docker-compose logs -f wordpress`
+`docker-compose up -d`
 
-which is useful because it gives you WP log output.
+This will build docker container and image with the specified settings from the `.yml` file in the detached state.
```

Comment:
This will build a Docker container and image with the specified settings from the `.yml` file in the detached state.